src/System.IO.Abstractions.Extensions/IDirectoryInfoExtensions.cs

```diff
@@ -3,25 +3,103 @@
         {
-            return info.FileSystem.DirectoryInfo.New(info.FileSystem.Path.Combine(info.FullName, name));
+            return info.CopyTo(destDirectoryName, false);
+        }
+
+        /// <summary>
+        ///     Copies this <see cref="IDirectoryInfo" /> instance and its contents to a new path.
+        /// </summary>
+        /// <param name="info"></param>
+        /// <param name="destDirectoryName">The name and path to wich to copy this directory. The destination must not exist.</param>
+        /// <param name="recursive"><see langword="true" /> to copy this directory, its subfolders, and all files; otherwise <see langword="false" />.</param>
+        /// <returns>An <see cref="IDirectoryInfo" /> for the specified path.</returns>
+        public static IDirectoryInfo CopyTo(this IDirectoryInfo info, string destDirectoryName, bool recursive)
+        {
+            var dest = info.FileSystem.DirectoryInfo.New(destDirectoryName);
+            return info.CopyTo(dest, recursive);
         }
 
         /// <summary>
-        /// Get an <see cref="IFileInfo"/> for the specified file <paramref name="name"/>
+        ///     Copies this <see cref="IDirectoryInfo" /> instance and its contents to a new path.
+        /// </summary>
+        /// <param name="info"></param>
+        /// <param name="destDirectory">The <see cref="IDirectoryInfo" /> to wich to copy this directory. The destination must not exist.</param>
+        /// <returns>An <see cref="IDirectoryInfo" /> for the specified path.</returns>
+        public static IDirectoryInfo CopyTo(this IDirectoryInfo info, IDirectoryInfo destDirectory)
+        {
+            return info.CopyTo(destDirectory, false);
+        }
+
+        /// <summary>
+        ///     Copies this <see cref="IDirectoryInfo" /> instance and its contents to a new path.
+        /// </summary>
+        /// <param name="info"></param>
+        /// <param name="destDirectory">The <see cref="IDirectoryInfo" /> to wich to copy this directory. The destination must not exist.</param>
+        /// <param name="recursive"><see langword="true" /> to copy this directory, its subfolders, and all files; otherwise <see langword="false" />.</param>
+        /// <returns>An <see cref="IDirectoryInfo" /> for the specified path.</returns>
+        public static IDirectoryInfo CopyTo(this IDirectoryInfo info, IDirectoryInfo destDirectory, bool recursive)
+        {
+            info.Refresh();
+            if (!info.Exists)
+            {
+                throw new DirectoryNotFoundException($"Source directory not found: '{info.FullName}'");
+            }
+
+            destDirectory.Refresh();
+            if (destDirectory.Exists)
+            {
+                throw new IOException($"The directory '{destDirectory.FullName}' already exists.");
+            }
+
+            var fileSystem = info.FileSystem;
+            fileSystem.Directory.CreateDirectory(destDirectory.FullName);
```

Comment:
I might be wrong but I think a create method exists like destDirectory.Create() or not?

Reply:
I thought destDirectory.Create() would not create the hierarchy if a parent does not exist. With destDirectory.Create() it looks much nicer.